src/_pytest/tmpdir.py

```diff
@@ -26,7 +27,14 @@ class TempPathFactory(object):
 
     The base directory can be configured using the ``--basetemp`` option."""
 
-    _given_basetemp = attr.ib()
+    _given_basetemp = attr.ib(
+        # using os.path.abspath() to get absolute path instead of resolve() as it
+        # does not work the same in all platforms (see #4427)
+        # Path.absolute() exists, but it is not public (see https://bugs.python.org/issue25012)
+        convert=attr.converters.optional(
+            lambda p: Path(os.path.abspath(six.text_type(p)))
```

Comment:
Brief note (was stalking IRC so I saw this comment) -- `resolve()` doesn't only resolve symlinks, but on windows it also resolves UNC paths from mounts (e.g. if you mount a network share at `R:\\Some\\path` but it really points at `\\10.1.100.100\Some\path`, `Path.resolve()` captures that but `os.path.abspath()` will return `R:\\blah` -- despite `absolute` not being public we typically handle this by trying `resolve` and falling back on `absolute` in pipenv at least

Reply:
Thanks @techalchemy for the note, that's interesting! 👍